### PR TITLE
Auth update

### PR DESCRIPTION
**Pull request by toxic-bishop**
- [x] Enable User Auto Login
- [x] Persist User prefs
- [x] Create User Service to get user details

### CODE REVIEW

File: lib/models/preferences_model/language_and_region/language_and_region.dart [hunk 1]
```diff
@@ -1,14 +1,18 @@
 // import 'package:freezed_annotation/freezed_annotation.dart';
-// import 'package:zc_desktop_flutter/ui/shared/preferences/preferenceswidgets/advanced/advanced_viewmodel.dart';
 
 // part 'language_and_region.freezed.dart';
 // part 'language_and_region.g.dart';
 
 // @freezed
 // class LanguageAndRegion with _$LanguageAndRegion {
 //   factory LanguageAndRegion({
+//     //@Default('English') String? languages,
```
Comment (isaacadariku) [inline lib/models/preferences_model/language_and_region/language_and_region.dart:9]: The file should be removed, if nothing is needed

File: lib/ui/auth/login_page/login_viewmodel.dart [hunk 2]
```diff
@@ -48,13 +49,11 @@ class LoginViewModel extends BaseViewModel {
...
-
-    // } catch (e) {
-    await Future.delayed(Duration(milliseconds: 1000));
-    _navigationService.navigateTo(Routes.loginView);
-    // }
+    try {
+      await _user.fetchAndSetUserData();
+      _navigationService.navigateTo(Routes.workspaceView);
+    } catch (e) {
+      _navigationService.navigateTo(Routes.loginView);
```
Comment (isaacadariku) [inline lib/ui/auth/login_page/login_viewmodel.dart:56]: We don't need to navigate again, It's better we have a toast that tells the use couldn't login.

File: lib/ui/shared/const_app_colors.dart [hunk 3]
```diff
@@ -50,4 +50,8 @@ const Color authBtnColor = Color(0xff363636);
 const Color preferenceBorderblack = Color(0xFF323232);
 
 //status colors
-const Color clearStatusBtnBorderColor = Color(0xffa1a1a1);
\ No newline at end of file
+const Color clearStatusBtnBorderColor = Color(0xffa1a1a1);
+
+//Preferences colors
+const Color kcPrefTitleColor = Color.fromRGBO(29, 29, 29, 1);
```
Comment (isaacadariku) [inline lib/ui/shared/const_app_colors.dart:56]: Why is preferences having it's own color, is the app colors not general

File: lib/ui/shared/const_text_styles.dart [hunk 4]
```diff
@@ -223,3 +211,33 @@ final clearStatusTextStyle = TextStyle(
   color: bodyColor,
   fontFamily: 'Lato',
 );
+
+//Preferences TextStyle
+
+final leftSideBarPrefTextStyle = TextStyle(
```
Comment (isaacadariku) [inline lib/ui/shared/const_text_styles.dart:217]: Also for the textstyles below

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart [hunk 5]
```diff
@@ -245,7 +413,36 @@ class NotificationView extends StatelessWidget {
       }),
     ]);
   }
-}
 
 
+  
 
+  headerSpacer() {
```
Comment (isaacadariku) [inline lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart:420]: This is not valid

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart [hunk 6]
```diff
@@ -245,7 +413,36 @@ class NotificationView extends StatelessWidget {
...
+    return SizedBox(
+      height: 3,
+    );
+  }
+
+  buildHeaderAndSubTitle(String header, String subheader) {
```
Comment (isaacadariku) [inline lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart:426]: This is not valid too, no functional widget

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart [hunk 7]
```diff
@@ -245,7 +413,36 @@ class NotificationView extends StatelessWidget {
...
+        headerSpacer(),
+        Text(
+          'Choose your notification sound.',
+          style: prefSubTitleTextStyle,
+        ),
+      ],
+    );
+  }
+
+  buildDivider() {
```
Comment (isaacadariku) [inline lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart:442]: This too

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_viewmodel.dart [hunk 8]
```diff
@@ -1,8 +1,16 @@
...
 import 'package:stacked/stacked.dart';
+import 'package:zc_desktop_flutter/app/app.locator.dart';
 import 'package:zc_desktop_flutter/core/enums/flash_windows.dart';
 import 'package:zc_desktop_flutter/core/enums/pref_message.dart';
+import 'package:zc_desktop_flutter/models/preferences_model/notification/notification.dart';
+import 'package:zc_desktop_flutter/services/local_storage/local_storage_service.dart';
 
 class NotificationViewModel extends BaseViewModel {
+  final _localStorage = locator<LocalStorageService>();
+  final _notificationStorageKey = 'notificationSetting';
```
Comment (isaacadariku) [inline lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_viewmodel.dart:11]: Why are you defining the key here in the viewmodel, move it to the sharedPrefrences set up to keep consistency

File: lib/ui/shared/preferences/preferenceswidgets/theme/theme_viewmodel.dart [hunk 9]
```diff
@@ -36,8 +40,10 @@ enum cleanThemes { aubergine, versatile, aubergine1, aubergine2 }
 
 class ThemeViewModel extends BaseViewModel {
   final log = getLogger("ThemeViewModel");
-
   final _themeService = locator<ThemeService>();
+  final _localStorage = locator<LocalStorageService>();
+  final _themeStorageKey = 'themeSetting';
```
Comment (isaacadariku) [inline lib/ui/shared/preferences/preferenceswidgets/theme/theme_viewmodel.dart:45]: This too, all keys should be setup in the localStorage file

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Nice I can see your improvement, Welldone... Go throught the changes requested and fix them according then we are good to go 👍🏽

File: lib/models/preferences_model/language_and_region/language_and_region.dart (hunk 1 above)
Comment (toxic-bishop) [reply lib/models/preferences_model/language_and_region/language_and_region.dart]: removed

File: lib/ui/shared/const_app_colors.dart (hunk 3 above)
Comment (toxic-bishop) [reply lib/ui/shared/const_app_colors.dart]: done

File: lib/ui/shared/const_text_styles.dart (hunk 4 above)
Comment (toxic-bishop) [reply lib/ui/shared/const_text_styles.dart]: done

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart (hunk 6 above)
Comment (toxic-bishop) [reply lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart]: done

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart (hunk 5 above)
Comment (toxic-bishop) [reply lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart]: done

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart (hunk 7 above)
Comment (toxic-bishop) [reply lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_view.dart]: done

File: lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_viewmodel.dart (hunk 8 above)
Comment (toxic-bishop) [reply lib/ui/shared/preferences/preferenceswidgets/notification/preferences_notification_viewmodel.dart]: done

File: lib/ui/shared/preferences/preferenceswidgets/theme/theme_viewmodel.dart (hunk 9 above)
Comment (toxic-bishop) [reply lib/ui/shared/preferences/preferenceswidgets/theme/theme_viewmodel.dart]: done

File: lib/ui/auth/login_page/login_viewmodel.dart (hunk 2 above)
Comment (toxic-bishop) [reply lib/ui/auth/login_page/login_viewmodel.dart]: Solved.

**Review by isaacadariku — DISMISSED** [review]
LGTM, we will still review later, but this is fine for now.

**Comment by isaacadariku** [comment]
@toxic-bishop fix conflicts so it can be merge

**Pull request closed without merging**